metadata-ingestion/src/datahub/classification/classifier_pipeline.py

```diff
@@ -0,0 +1,278 @@
+
+import click
+import requests
+
+from datahub.classification.classifier import ClassificationResult, Classifier
+from datahub.configuration.common import ConfigModel, DynamicTypedConfig
+from datahub.ingestion.api.common import PipelineContext, RecordEnvelope
+from datahub.ingestion.api.sampleable_source import SampleableSource
+from datahub.ingestion.api.source import Extractor
+from datahub.ingestion.api.transform import Transformer
+from datahub.ingestion.extractor.extractor_registry import extractor_registry
+from datahub.ingestion.source.source_registry import source_registry
+from datahub.ingestion.transformer.transform_registry import transform_registry
+from datahub.metadata.schema_classes import SchemaMetadataClass
+
+logger = logging.getLogger(__name__)
+
+
+class SourceConfig(DynamicTypedConfig):
+    extractor: str = "generic"
+
+
+class ClassifierPipelineConfig(ConfigModel):
+    datahub_base_url: str
+    datahub_username: str
+    datahub_encrypted_password: str
+    ddb_client: Any  # Type hints not available for boto3 clients
+    kms_client: Any  # Type hints not available for boto3 clients
+    num_shards: int
+    pii_classification_state_table_name: str
+    shard_id: int
+    source: SourceConfig
+    transformers: Optional[List[DynamicTypedConfig]]
+
+
+class ClassifierPipeline:
+    DATAHUB_GRAPHQL_ENDPOINT: str = "/api/v2/graphql"
+    DATAHUB_LOGIN_ENDPOINT: str = "/logIn"
+    DATASET_RECORD_SK: str = "DATASET"
+
+    config: ClassifierPipelineConfig
+    ctx: PipelineContext
+    classifier: Classifier
+    datahub_base_url: str
+    datahub_cookies: Dict[str, str]
+    my_shard_id: int
+    num_shards: int
+    source: SampleableSource
+    transformers: List[Transformer]
+
+    def __init__(self, config: ClassifierPipelineConfig):
+        logger.info(f"Building classifier pipeline from config: {config}")
+        self.config = config
+        self.ctx = PipelineContext(run_id=str(uuid.uuid1()))
+        source_type = self.config.source.type
+        source_class = source_registry.get(source_type)
+        self.source: SampleableSource = source_class.create(
+            self.config.source.dict().get("config", {}), self.ctx
+        )
+        self.extractor_class = extractor_registry.get(self.config.source.extractor)
+
+        self.ddb_client = config.ddb_client
+        self.kms_client = config.kms_client
+        self.classifier = Classifier(self.source)
+        self.pii_classification_state_table_name = (
+            config.pii_classification_state_table_name
+        )
+        self.datahub_base_url = config.datahub_base_url
+        self.datahub_cookies = self._login_to_datahub(
+            config.datahub_username, config.datahub_encrypted_password
+        )
+        self.my_shard_id = config.shard_id
+        self.num_shards = config.num_shards
+        assert self.my_shard_id >= 0 and self.my_shard_id < self.num_shards
+
+        self._configure_transforms()
+
+    def _login_to_datahub(
```

Comment:
minor: this could probably be taken out to a higher level and be resued if other things need to login?

Reply:
Possibly, might wait until we cross that bridge though

Reply:
Sounds good to me.